sklearn/sklearn-template/template/trainer/model.py

```diff
@@ -20,34 +20,48 @@
 from sklearn import pipeline
 from sklearn import preprocessing
 
+import numpy as np
+
 from trainer import metadata
 
 
 def get_estimator(flags):
-  classifier = ensemble.RandomForestClassifier()
-
-  # TODO(cezequiel): Make use of flags for hparams
-  _ = flags
+  # TODO: Allow pre-processing to be configurable through flags
+  classifier = ensemble.RandomForestClassifier(**flags)
 
   numeric_transformer = pipeline.Pipeline([
-      ('imputer', impute.SimpleImputer(strategy='median')),
-      ('scaler', preprocessing.StandardScaler()),
+    ('imputer', impute.SimpleImputer(strategy='median')),
+    ('scaler', preprocessing.StandardScaler()),
+  ])
+
+  numeric_log_transformer = pipeline.Pipeline([
+    ('imputer', impute.SimpleImputer(strategy='median')),
+    ('log', preprocessing.FunctionTransformer(
+      func=np.log1p, inverse_func=np.expm1, validate=True)),
+    ('scaler', preprocessing.StandardScaler()),
+  ])
+
+  numeric_bin_transformer = pipeline.Pipeline([
+    ('imputer', impute.SimpleImputer(strategy='median')),
+    ('bin', preprocessing.KBinsDiscretizer(n_bins=5, encode='onehot-dense')),
   ])
 
   categorical_transformer = pipeline.Pipeline([
-      ('imputer', impute.SimpleImputer(
-          strategy='constant', fill_value='missing')),
-      ('onehot', preprocessing.OneHotEncoder(handle_unknown='ignore')),
+    ('imputer', impute.SimpleImputer(
+      strategy='constant', fill_value='missing')),
+    ('onehot', preprocessing.OneHotEncoder(handle_unknown='ignore', sparse=False)),
   ])
 
   preprocessor = compose.ColumnTransformer([
-      ('numeric', numeric_transformer, metadata.NUMERIC_FEATURES),
-      ('categorical', categorical_transformer, metadata.CATEGORICAL_FEATURES),
+    ('numeric', numeric_transformer, metadata.NUMERIC_FEATURES),
```

Comment:
You might need to replace `numeric` with some other column transform name, otherwise they the transforms might overlap. Were you able to verify the behavior?

Reply:
Sorry, forgot to change the name while copy-pasting ....

Reply:
No worries.